app/(api)/_datalib/events/deleteEvent.ts

```diff
@@ -0,0 +1,33 @@
+import { getDatabase } from '@utils/mongodb/mongoClient.mjs';
+import { HttpError, NotFoundError } from '@utils/response/Errors';
+import { ObjectId } from 'mongodb';
+
+export async function deleteEvent(id: string) {
+  try {
+    const db = await getDatabase();
+    const objectId = ObjectId.createFromHexString(id);
+
+    const deletion = await db.collection('events').deleteOne({
+      _id: objectId,
+    });
+
+    if (deletion.deletedCount === 0) {
+      throw new NotFoundError(
+        `Could not delete event with ID: '${id}'. Event does not exist or ID is incorrect.`
+      );
+    }
+
+    return {
+      ok: true,
+      body: null,
```

Comment:
Change body to "Event deleted." for consistency in return type: 
`{ok: boolean, body: string | object, error: null} || {ok: boolean, body: null, error: string}`